File: frontend/src/features/auth/services/authService.ts
import { apiClient } from '@/lib/apiClient';
import type { LoginCredentials, AuthResponse } from '@/features/auth/types';

/**
 * Service to handle authentication related API calls.
 * Follows the Separation of Concerns principle.
 */
export const authService = {
  /**
   * authenticates a user and returns tokens/user data.
   */
  login: async (credentials: LoginCredentials): Promise<AuthResponse> => {
    const response = await apiClient.post<AuthResponse>('/auth/login', credentials);
    return response.data;
  },

  /**
   * Refreshes the access token using the HttpOnly cookie.
   */
  refresh: async (): Promise<{ accessToken: string }> => {
    const response = await apiClient.post<{ accessToken: string }>('/auth/refresh');
    return response.data;
  },

  /**
   * Log out the user remotely (if needed).
   */
  logout: async (): Promise<void> => {
    await apiClient.post('/auth/logout');
  }
};

File: file_vault_app/lib/features/auth/auth_service.dart
import 'package:dio/dio.dart';
import 'package:file_vault_app/core/api/api_client.dart';
import 'package:file_vault_app/features/auth/auth_model.dart';

/// Handles authentication API calls.
/// Calls POST /auth/login and parses the response.
class AuthService {
  final Dio _dio = ApiClient.instance;

  /// Login with email and password.
  /// Returns (accessToken, user) on success.
  /// Throws DioException on failure (401 = invalid credentials, etc).
  Future<(String, AuthUser)> login({
    required String email,
    required String password,
  }) async {
    final response = await _dio.post(
      '/auth/login',
      data: {'email': email, 'password': password},
    );

    // Backend wraps response: { success: true, data: { accessToken, user } }
    final data = response.data['data'] as Map<String, dynamic>;
    final accessToken = data['accessToken'] as String;
    final user = AuthUser.fromJson(data['user'] as Map<String, dynamic>);

    return (accessToken, user);
  }
}
